src/main/scala-2.12/collection/Factory.scala

```diff
@@ -0,0 +1,36 @@
+package scala.collection
+
+import scala.collection.generic.CanBuildFrom
+
+/**
+  * A factory that builds a collection of type `C` with elements of type `A`.
+  *
+  * @tparam A Type of elements (e.g. `Int`, `Boolean`, etc.)
+  * @tparam C Type of collection (e.g. `List[Int]`, `TreeMap[Int, String]`, etc.)
+  */
+trait Factory[-A, +C] extends Any {
+
+  /**
+    * @return A collection of type `C` containing the same elements
+    *         as the source collection `it`.
+    * @param it Source collection
+    */
+  def fromSpecific(it: TraversableOnce[A]): C
+
+  /** Get a Builder for the collection. For non-strict collection types this will use an intermediate buffer.
+    * Building collections with `fromSpecific` is preferred because it can be lazy for lazy collections. */
+  def newBuilder(): mutable.Builder[A, C]
+}
+
+object Factory {
+
+  implicit def fromCanBuildFrom[A, C](implicit cbf: CanBuildFrom[Nothing, A, C]): Factory[A, C] =
+    new Factory[A, C] {
+      def fromSpecific(it: TraversableOnce[A]): C = (cbf() ++= it).result()
```

Comment:
This needs special treatment of `Stream` and `LazyList` to preserve laziness.

Reply:
Actually this works because `Stream` uses a `LazyBuilder`. I’ve added a test for that.